index.js

```diff
@@ -79,7 +79,17 @@ function isWorkerSupported() {
         return false;
     }
 
-    var blob = new Blob([''], { type: 'text/javascript' });
+    // Safari 5.1 has a bug where window.Blob is not falsy
+    var blob;
+    try {
+      blob = new Blob([''], { type: 'text/javascript' });
+    } catch (e) {
+      if (e instanceof TypeError) {
```

Comment:
Since we're handling a very specific compatibility issue here, let's make this check more specific -- i.e., check for `[object BlobConstructor]' is not a constructor` being part of the error message.

Also, could you please attach the comment above to this line instead?

Reply:
Can we do the check without a `try`/`catch`? 

Something along these lines may work: 

```js
 if (typeof(Blob) === 'function') {
```

Reply:
@anandthakker ok

@lucaswoj I tried that after seeing similar suggestion for this bug, but that test passes too on Safari 5.1